examples/update_bids_datasets.py

```diff
@@ -0,0 +1,137 @@
+do not have all the metadata stored in the ``Raw`` mne-python object.
+We can update the BIDS sidecar files via the ``update_sidecar_json`` function.
+
+In this tutorial, we show how ``update_sidecar_json`` can be used to update and
+modify BIDS-formatted data.
+"""
+# Authors: Adam Li <adam2392@gmail.com>
+#          mne-bids developers
+#
+# License: BSD (3-clause)
+
+###############################################################################
+# .. contents:: Contents
+#    :local:
+#    :depth: 1
+#
+# Imports
+# -------
+# We are importing everything we need for this example:
+from mne.datasets import somato
+
+from mne_bids import (BIDSPath, read_raw_bids,
+                      print_dir_tree, make_report, update_sidecar_json)
+
+###############################################################################
+# We will be using the `MNE somato data <mne_somato_data_>`_, which
+# is already stored in BIDS format.
+# For more information, you can check out the
+# respective :ref:`example <ex-convert-mne-sample>`.
+#
+# Download the ``somato`` BIDS dataset
+# ------------------------------------
+#
+# Download the data if it hasn't been downloaded already, and return the path
+# to the download directory. This directory is the so-called `root` of this
+# BIDS dataset.
+bids_root = somato.data_path()
+
+###############################################################################
+# Explore the dataset contents
+# ----------------------------
+#
+# We can use MNE-BIDS to print a tree of all
+# included files and folders. We pass the ``max_depth`` parameter to
+# `mne_bids.print_dir_tree` to the output to three levels of folders, for
+# better readability in this example.
+
+print_dir_tree(bids_root, max_depth=3)
+
+# We can generate a report of the existing dataset
+print(make_report(bids_root))
+
+###############################################################################
+# Update the sidecar JSON dataset contents
+# ----------------------------------------
+#
+# We can use MNE-BIDS to update all sidecar files for a matching
+# ``BIDSPath`` object. We then pass in a dictionary (or JSON file) to update
+# all matching metadata fields within the BIDS dataset.
+
+# create a BIDSPath object
+bids_root = somato.data_path()
+datatype = 'meg'
+subject = '01'
+task = 'somato'
+suffix = 'meg'
+
+bids_path = BIDSPath(subject=subject, task=task, suffix=suffix,
+                     datatype=datatype, root=bids_root)
+sidecar_path = bids_path.copy().update(extension='.json')
+
+# We can now retrieve a list of all MEG-related files in the dataset:
+print(bids_path.match())
+
+# Define a sidecar update as a dictionary
+entries = {
+    'PowerLineFrequency': 60,
+    'Manufacturer': "MEGIN",
+    'InstitutionName': "Martinos Center"
+}
+
+# Note: ``update_sidecar_json`` will perform essentially a
+# dictionary update to your sidecar file, so be absolutely sure
+# that the ``entries`` are defined properly.
```

Comment:
I'm not really sure what this is supposed to tell me?

Reply:
I've updated this description to just warn users that they want to be careful when updating since we don't do any checks based on valid fields per sidecar json file. It'll assume ``entries`` is defined correctly and properly for the corresponding `bids_path`.